### PR TITLE
PS-8048: Merge MySQL 5.7.37 part 1 (up to d773112b)

### PR DESCRIPTION
**Pull request by nitendra-bhosle**
https://jira.percona.com/browse/PS-8048

***
    VERSION raised to "5.7.37-40.0".
    PERCONA_INNODB_VERSION in univ.i raised to "40.0".

***
Updated man pages from MySQL Server 5.7.37 source tarball.
***
Updated 'scripts/fill_help_tables.sql' from MySQL Server 5.7.37 source tarball.

### CODE REVIEW

**Comment by nitendra-bhosle** [comment]
Jenkins : https://ps57.cd.percona.com/view/5.7/job/percona-server-5.7-param/530/

File: MYSQL_VERSION [hunk 1]
```diff
@@ -1,12 +1,4 @@
...
-MYSQL_VERSION_PATCH=36
-MYSQL_VERSION_EXTRA=-39
-||||||| 0ed6d65f4c6:VERSION
-MYSQL_VERSION_PATCH=36
-MYSQL_VERSION_EXTRA=
-=======
 MYSQL_VERSION_PATCH=37
-MYSQL_VERSION_EXTRA=
->>>>>>> d773112b3a819360faa71023b147c0ecd8bc90e6:VERSION
+MYSQL_VERSION_EXTRA=-41
```
Comment (oleksandr-kachan) [inline MYSQL_VERSION]: Previous MYSQL_VERSION_EXTRA was 39, it should be just incremented

File: include/mysql_com.h [hunk 2]
```diff
@@ -137,13 +137,9 @@
 #define FIELD_IS_DROPPED (1<< 26)       /* Intern: Field is being dropped */
 #define EXPLICIT_NULL_FLAG (1<< 27)     /* Field is explicitly specified as
                                            NULL by the user */
-<<<<<<< HEAD
 #define CLUSTERING_FLAG (1 << 28)       /* Field has a secondary clustering
                                         key */
-||||||| 0ed6d65f4c6
-=======
 #define SERIAL_FLAG (1<< 28)            /* Field is "SERIAL" */
```
Comment (oleksandr-kachan) [inline include/mysql_com.h:140]: We need to update bit offset here for it not to overlap with ours CLUSTERING_FLAG

File: storage/innobase/include/univ.i [hunk 3]
```diff
@@ -55,7 +55,7 @@ Created 1/20/1994 Heikki Tuuri
 #define INNODB_VERSION_BUGFIX	MYSQL_VERSION_PATCH
 
 #ifndef PERCONA_INNODB_VERSION
-#define PERCONA_INNODB_VERSION 39
+#define PERCONA_INNODB_VERSION 41
```
Comment (oleksandr-kachan) [inline storage/innobase/include/univ.i]: 40

Comment (oleksandr-kachan) [reply storage/innobase/include/univ.i]: Please fix version in commit description as well

**Comment by oleksandr-kachan** [comment]
Please also update PR name and include version changes info in its description. See https://github.com/percona/percona-server/pull/4513 for an example.

**Review by oleksandr-kachan — CHANGES_REQUESTED** [review]
Please see comments above

**Comment by nitendra-bhosle** [comment]
Jenkins : https://ps57.cd.percona.com/view/5.7/job/percona-server-5.7-param/536/
jenkins ASAN : https://ps57.cd.percona.com/job/percona-server-5.7-asan-param/41/

**Comment by nitendra-bhosle** [comment]
Valgrid : https://ps57.cd.percona.com/job/percona-server-5.7-pipeline/25815/

File: mysql-test/r/mysql_config_editor.result [hunk 4]
```diff
@@ -100,14 +100,8 @@ user = test_user4
 ##############################################
 # Tests for mysql_config_editor's help command
 ##############################################
-<<<<<<< HEAD
 Copyright (c) 2009-2021 Percona LLC and/or its affiliates
```
Comment (oleksandr-kachan) [inline mysql-test/r/mysql_config_editor.result]: Shall we change year to 2009-2022 here?

Comment (oleksandr-kachan) [reply mysql-test/r/mysql_config_editor.result]: Ok, I see this is done in next commit. Would it be better to move this change to "test fixes" then?

**Comment by oleksandr-kachan** [comment]
Please fix commit message for
https://github.com/percona/percona-server/pull/4614/commits/8c144e4af632bc1b9192058839c43d1b798db136 a bit. No reason to mention rebase any more.

File: include/m_string.h [hunk 5]
```diff
@@ -28,15 +28,6 @@
 
 #include <string.h>
 
-<<<<<<< HEAD
```
Comment (kamil-holubicki) [inline include/m_string.h]: Is it OK? For some reason, PS deleted this part.

Comment (nitendra-bhosle) [reply include/m_string.h]: I think , this should be OK. 
@oleksandr-kachan  what do you say ?

Comment (oleksandr-kachan) [reply include/m_string.h]: This is Ok. This was our fix for OSx build issues. Upstream now fixed same thing in a different way so we can just use their version.

**Review by kamil-holubicki — APPROVED** [review]
lgtm

File: include/mysql_com.h [hunk 6]
```diff
@@ -137,13 +137,9 @@
...
 #define EXPLICIT_NULL_FLAG (1<< 27)     /* Field is explicitly specified as
                                            NULL by the user */
-<<<<<<< HEAD
 #define CLUSTERING_FLAG (1 << 28)       /* Field has a secondary clustering
                                         key */
-||||||| 0ed6d65f4c6
-=======
-#define SERIAL_FLAG (1<< 28)            /* Field is "SERIAL" */
->>>>>>> 0cd98bdf981583a1cf4cb526581fc16e23bb839b^
+#define SERIAL_FLAG (1<< 29)            /* Field is "SERIAL" */
```
Comment (dutow) [inline include/mysql_com.h]: Based on `testclient/mysql_client_test.c`, it looks like these flags are sent to the client.

I see that right now the client doesn't do anything with this flag, but I'm still wondering if this is a good choice here.

Anytime PS sets the CLUSTERING_FLAG, the oracle client will see a field set with the SERIAL_FLAG. (or a custom client based on libmysql)

Unless I missed something in the code, we do not write these flags to the disk, so we could easily change the id of CLUSTERING_FLAG instead?

Comment (nitendra-bhosle) [reply include/mysql_com.h]: Yeah sure. If this is the case ,we can keep #define SERIAL_FLAG (1<< 28)  as is and increment ID of CLUSTERING_FLAG
as  #define CLUSTERING_FLAG (1 << 29)
@oleksandr-kachan  are you ok with this?

Comment (nitendra-bhosle) [reply include/mysql_com.h]: Done.

**Review by dutow — APPROVED** [review]

**Comment by nitendra-bhosle** [comment]
Final Jenkins run : https://ps57.cd.percona.com/view/5.7/job/percona-server-5.7-param/550/ 
No new failures .

**Review by dutow — APPROVED** [review]

**Pull request merged**